testkit/src/checkpoint_db.rs

```diff
@@ -12,38 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use exonum_merkledb::{Database, Patch, Result as StorageResult, Snapshot, View};
-
-use std::sync::{Arc, RwLock};
+use exonum_merkledb::{Database, Patch, Result as StorageResult, Snapshot};
+use rand::{
+    distributions::{Alphanumeric, Distribution},
+    thread_rng,
+};
+use tempfile::TempDir;
+
+use std::{
+    path::Path,
+    sync::{Arc, RwLock},
+};
 
 /// Implementation of a `Database`, which allows to rollback its state
 /// to the last made checkpoint.
 ///
 /// **Note:** Intended for testing purposes only. Probably inefficient.
 #[derive(Debug)]
-pub struct CheckpointDb<T> {
-    inner: Arc<RwLock<CheckpointDbInner<T>>>,
+pub struct CheckpointDb {
```

Comment:
Are the changes in `CheckpointDb` caused by the desire to test checkpoint functionality? Or is there demand to run `TestKit` with the persistent database? In the first case, the changes look like an overkill to me (the old implementation is still functional, isn't it?).